Engine/gfx/blender.h

```diff
@@ -62,4 +62,109 @@ void set_opaque_alpha_blender();
+#define _BLENDOP_DARKEN(B,L) ((L > B) ? B:L)
+#define _BLENDOP_EXCLUSION(B,L) (B + L - 2 * B * L / 255)
+#define _BLENDOP_SUBTRACT(B,L) (L - B < 0) ? 0:(L - B)
+// not very pretty but forces the original alpha of "x" to "blender_result"
+inline unsigned long _blender_mask_alpha24(unsigned long blender_result, unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long src_alpha = geta32(x);
+    const unsigned long alphainv = 255 - src_alpha;
+    const unsigned long r = (getr24(blender_result)*src_alpha + alphainv*getr24(y)) / 255;
+    const unsigned long g = (getg24(blender_result)*src_alpha + alphainv*getg24(y)) / 255;
+    const unsigned long b = (getb24(blender_result)*src_alpha + alphainv*getb24(y)) / 255;
+    return r | g << 8 | b << 16;
+}
+// alegro's dodge blend was wrong, mine is not perfect either
+inline unsigned long _my_blender_dodge24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = (getr24(y)*n) / (256 - getr24(x));
+    const unsigned long j = (getg24(y)*n) / (256 - getg24(x));
+    const unsigned long k = (getb24(y)*n) / (256 - getb24(x));
+    return BLEND(24, h>255 ? 255 : h, j>255 ? 255 : j, k>255 ? 255 : k);
+}
+inline unsigned long _my_blender_burn24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = _BLENDOP_BURN(getr24(x), getr24(y));
+    const unsigned long j = _BLENDOP_BURN(getg24(x), getg24(y));
+    const unsigned long k = _BLENDOP_BURN(getb24(x), getb24(y));
+    return BLEND(24, h, j, k);
+}
+inline unsigned long _my_blender_lighten24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = _BLENDOP_LIGHTEN(getr24(x), getr24(y));
+    const unsigned long j = _BLENDOP_LIGHTEN(getg24(x), getg24(y));
+    const unsigned long k = _BLENDOP_LIGHTEN(getb24(x), getb24(y));
+    return BLEND(24, h, j, k);
+}
+inline unsigned long _my_blender_darken24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = _BLENDOP_DARKEN(getr24(x), getr24(y));
+    const unsigned long j = _BLENDOP_DARKEN(getg24(x), getg24(y));
+    const unsigned long k = _BLENDOP_DARKEN(getb24(x), getb24(y));
+    return BLEND(24, h, j, k);
+}
+inline unsigned long _my_blender_exclusion24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = _BLENDOP_EXCLUSION(getr24(x), getr24(y));
+    const unsigned long j = _BLENDOP_EXCLUSION(getg24(x), getg24(y));
+    const unsigned long k = _BLENDOP_EXCLUSION(getb24(x), getb24(y));
+    return BLEND(24, h, j, k);
+}
+inline unsigned long _my_blender_subtract24(unsigned long x, unsigned long y, unsigned long n)
+{
+    const unsigned long h = _BLENDOP_SUBTRACT(getr24(x), getr24(y));
+    const unsigned long j = _BLENDOP_SUBTRACT(getg24(x), getg24(y));
+    const unsigned long k = _BLENDOP_SUBTRACT(getb24(x), getb24(y));
+    return BLEND(24, h, j, k);
+}
+
+inline unsigned long _blender_masked_add32(unsigned long x, unsigned long y, unsigned long n)
```

Comment:
This is not critical, but I'd like to note that defining these functions (_blender_masked_add32 and so on) inlined in header makes little sense, as they are passed into sprite drawing routine as pointers, and code inlining will not happen anyway.